### ISSUE TITLE
Translate tool doesn't accept length multipliers in LOBs and CHARACTERS or OCTETS

### ISSUE THREAD

**Issue by katzyn** [jOOQ/jOOQ#10378]
1. Translate tool can't parse standard LOB data types with length multipliers such as `CLOB(10K)` or `BLOB(1M)`. There are `K`, `M`, `G`, `T`, and `P` multipliers in the standard, where `K` is 1024 and so on.

2. Translate tool can't parse standard character string data types with `CHARACTERS` or `OCTETS` units, such as `CHARACTER VARYING(100 CHARACTERS)` or `CLOB(100000 OCTETS)`.

**Comment by lukaseder** [comment]
Thank you very much for your feature request. I must admit, I have not yet seen those multipliers in the wild yet. Will definitely be interesting to keep an eye on, in case someone needs this.

The `CHAR` vs `OCTETS` (or `BYTE` in Oracle) distinction should be implemented in the future, indeed. There's already a feature request for this: https://github.com/jOOQ/jOOQ/issues/2288

I'll edit this feature request to keep things separate.

**Comment by katzyn** [comment]
I saw only non-standard units from Oracle and `K` and `M` length multiplies (I don't remember where) in practice.

Multipliers aren't expected to be preserved by DMBS, I guess jOOQ may also support them only in the parser.

**Comment by lukaseder** [comment]
> I saw only non-standard units from Oracle and `K` and `M` length multiplies (I don't remember where) in practice.

Oh no, you mean, there are different interpretations *per dialect* of these multipliers? My work here will never end 😂

> Multipliers aren't expected to be preserved by DMBS, I guess jOOQ may also support them only in the parser.

Yes sure. This would be purely a parser feature

**Comment by katzyn** [comment]
No, I meant that I saw SQL scripts with non-standard `CHAR` and `BYTE`, but I didn't see standard `CHARACTERS` and `OCTETS` in use.

I also saw somewhere multipliers `K` and `M`.
